proposals/sets-tuples-dicts.md

```diff
@@ -0,0 +1,146 @@
+# Project Proposal: Dictionaries
+
+## 1. 10 Representative example programs related to our feature
+
+By integrating the support from strings and lists, we seek to support dictionary snippets such as: 
+
+
+**Example 1**: 

```

Comment:
There need to be some more sophisticated examples here. Not putting this as a blocker for a merge, but:

- How will type annotations look for dicts?
- All of these examples use constants in key/value position on literals, lookup, and assignment. How about some with interesting expressions there to make sure type-checking will work?

Reply:
@jpolitz okay yes, we'll update the examples in the proposal. 
Follow up question regarding type in dicts: Does this snippet need to throw an error in our compiler? In regular python in doesn't because after x = d[4] the type of x is int and not bool. But ideally in chocopy this should throw a type error.  
```
x:bool = True
d = {}
d[4] = 2
x = d[2+2]
```